### PR TITLE
Update Goal Confirmation

### LINKED ISSUE #1916: Add confirmation on Goal update
**Additional context**
This was previously implemented on `main` branch thanks to @Shubham-Praj , just has to be ported to `develop`.

**Is your feature request related to a problem? Please describe.**
It's very annoying when you don't get confirmation of your changes. After Goal update the edit modal just closes.

**Describe the solution you'd like**
1. A very brief Toast message 'Goal updated!'
2. A very short (compared to add Goal) sound
Not implemented in `main` - but should be implemented now: if there are no changes, don't show the modal.

**Describe alternatives you've considered**
A confirmation popup. This breaks the flow.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by Shubham-Praj**
Resolves #1916 

Once update goal is showing a toast message which says "Goal Updated!". And plays pling sound.

### CODE REVIEW

**Comment by tijlleenders** [comment]
@Shubham-Praj Is this ready for review? I was waiting for a review request.

**Comment by Shubham-Praj** [comment]
> @Shubham-Praj Is this ready for review? I was waiting for a review request.

Hi @tijlleenders, Yes, it is ready for review.

Could you please let me know if there is any further step after creating a PR ? like to initiate the process of review, or do we have to tag it ? 

I just created a pull request and added a resolves <issue id> in comments.

Thats it,like all the pull request which i had created previously.

**Comment by tijlleenders** [comment]
@Shubham-Praj Yes, you can request a review explicitly. Otherwise I don't know when you're done adding commits.  

Thanks for the PR, will review now!

**Review by tijlleenders — APPROVED** [review]
Looking good!

@Tushar-4781 Code ok?

File: src/components/GoalsComponents/GoalConfigModal/ConfigGoal.tsx [hunk 1]
```diff
@@ -149,7 +150,23 @@ const ConfigGoal = ({ goal, action }: { action: "Update" | "Create"; goal: GoalI
       }
       suggestChanges(rootGoal, getFinalTags(), title, goalColor, subGoalsHistory.length);
     } else {
-      await modifyGoal(goal.id, getFinalTags(), title, goalColor, [...ancestors, goal.id], hints);
+      updateStatus = await modifyGoal(goal.id, getFinalTags(), title, goalColor, [...ancestors, goal.id], hints);
+    }
+
+    if (updateStatus) {
```
Comment (Tushar-4781) [inline src/components/GoalsComponents/GoalConfigModal/ConfigGoal.tsx:156]: you can use ternary operator here as we need to show toast anyway

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]

File: src/components/GoalsComponents/GoalConfigModal/ConfigGoal.tsx (hunk 1 above)
Comment (Shubham-Praj) [reply src/components/GoalsComponents/GoalConfigModal/ConfigGoal.tsx:156]: Noted tushar I will fix this

**Comment by Shubham-Praj** [comment]
Hii @tijlleenders  this is ready for review and the code is updated as per @Tushar-4781 suggestion

**Comment by tijlleenders** [comment]
You can request a review from the github PR page here:
image

For this one I did it already.

Thanks for the good work!

**Comment by Shubham-Praj** [comment]
Noted @tijlleenders 
Thanks. I will raise the reviews from here in the future. 🫡

**Comment by tijlleenders** [comment]
@Tushar-4781 *bump*

**Comment by Tushar-4781** [comment]
hello @Shubham-Praj sorry for the inconvenience, but can you please checkout the **develop** branch and redo the PR.

**Comment by Shubham-Praj** [comment]
Hii @Tushar-4781 will check this and redo

**Comment by tijlleenders** [comment]
Merging this to main as well in the mean time. Thanks @Shubham-Praj !

**Review by tijlleenders — APPROVED** [review]

**Comment by Shubham-Praj** [comment]
Thanks @tijlleenders

**Pull request merged**